packages/kit/src/runtime/client/client.js

```diff
@@ -64,29 +64,37 @@ const scroll_positions = storage.get(SCROLL_KEY) ?? {};
  */
 const snapshots = storage.get(SNAPSHOT_KEY) ?? {};
 
-const original_push_state = BROWSER ? history.pushState : () => {};
-const original_replace_state = BROWSER ? history.replaceState : () => {};
-
 if (DEV && BROWSER) {
 	let warned = false;
 
 	const warn = () => {
 		if (warned) return;
+
+		// Avoid stashing away the original history methods which prevents monkeypatching by other libs,
+		// instead inspect the stack trace to see if we're being called from within SvelteKit.
+		let stack = new Error().stack?.split('\n');
+		if (!stack) return;
+		if (!stack[0].includes('https:') && !stack[0].includes('http:')) stack = stack.slice(1); // Chrome includes the error message in the stack
+		stack = stack.slice(2); // remove `warn` and the place where `warn` was called
+		if (stack[0].includes('src/runtime/client/client.js')) return;
```

Comment:
maybe this, just to make it slightly more resilient against file renames and whatnot? tested and it works

```suggestion
		if (stack[0].includes(import.meta.url)) return;
```